submissions/BlueLamantine/friends app/js/main.js

```diff
@@ -0,0 +1,329 @@
+    } else {
+      Array.from(cards).find(node => {
+        const currentGender = node.querySelector('.gender');
+        if (currentGender.textContent !== key) {
+          currentGender.closest('.user-card').classList.add('hidden');
+        } else {
+          currentGender.closest('.user-card').classList.remove('hidden');
+        }
+      });
+    }
+  }
+
+  insertSortedProfiles(sortedCards, param) {
+    sortedCards.forEach(card =>
+      this.profilesContainer.insertBefore(card, this.profilesContainer[param])
+    );
+  }
+
+  sortBySelector(selectorName) {
+    const cards = document.querySelectorAll('.user-card');
+    if (selectorName === 'all') {
+      return Array.from(cards).sort((a, b) => a.id.localeCompare(b.id));
+    }
+    return Array.from(cards).sort((a, b) =>
+      a
+        .querySelector(selectorName)
+        .textContent.localeCompare(b.querySelector(selectorName).textContent)
+    );
+  }
+
+  sortByContent(sortSelector, sortType, sortTypes) {
+    const sortedCards = this.sortBySelector(sortSelector);
+    const typeOfSort = {
+      [sortTypes.type1]: () =>
+        this.insertSortedProfiles(sortedCards, 'lastChild'),
+      [sortTypes.type2]: () =>
+        this.insertSortedProfiles(sortedCards, 'firstChild'),
+      [sortTypes.type3]: () =>
+        this.insertSortedProfiles(this.sortBySelector('all'), 'lastChild'),
+      default: () => false,
+    };
+    return (typeOfSort[sortType] || typeOfSort['default'])();
+  }
+}
+
+class Api {
+  constructor() {
+    this.apiURL = `https://randomuser.me/api/`;
+    this.errorMessage = `Uh oh, something has gone wrong. Please tweet us @randomapi about the issue. Thank you.`;
+  }
+
+  async getData(pageNum = 1) {
+    try {
+      const response = await fetch(
+        this.apiURL + `?page=${pageNum}&results=10&seed=abc`
+      );
+      if (!response.ok) throw new Error(errorMessage);
+      const data = await response.json();
+      new Render(data.results, data.info).getProfiles();
+      document.querySelector('.loading').classList.remove('active');
+    } catch (err) {
+      console.error(err);
+      alert(err);
+    }
+  }
+}
+
+class Observer {
+  constructor() {
+    this.startPage = 1;
+    this.loading = () => {
+      if (
+        window.scrollY + window.innerHeight >=
+        document.documentElement.scrollHeight - 5
+      ) {
+        document.querySelector('.loading').classList.add('active');
+        new Api().getData(++this.startPage);
+      }
+    };
+    document.querySelector('#menu').addEventListener('click', () => {
+      document.querySelector('#filters').classList.toggle('active');
+    });
+  }
+
+  loadDataByScroll(loadingStop) {
+    if (loadingStop) {
+      window.removeEventListener('scroll', this.loading);
+      return;
+    }
+    window.addEventListener('scroll', this.loading);
+  }
+}
+
+new Api().getData();
+new Filter();
+const loadMoreData = new Observer();
```

Comment:
Why class `Observer` has instance named `loadMoreData`?

Reply:
I removed from my app option to load more data with scroll. That's not necessary.